source/extensions/filters/network/rocketmq_proxy/topic_route.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RocketmqProxy {
+class QueueData {
+public:
+  QueueData(const std::string& brokerName, int32_t readQueueNum, int32_t writeQueueNum,
+            int32_t perm)
+      : broker_name_(brokerName), read_queue_nums_(readQueueNum), write_queue_nums(writeQueueNum),
+        perm_(perm) {}
+
+  void encode(ProtobufWkt::Struct& data_struct);
+
+  const std::string& brokerName() const { return broker_name_; }
+
+  int32_t readQueueNum() const { return read_queue_nums_; }
+
+  int32_t writeQueueNum() const { return write_queue_nums; }
+
+  int32_t perm() const { return perm_; }
+
+private:
+  std::string broker_name_;
+  int32_t read_queue_nums_;
+  int32_t write_queue_nums;
+  int32_t perm_;
+};
+
+class BrokerData {
+public:
+  BrokerData(const std::string& cluster, const std::string& brokerName,
+             std::unordered_map<int64_t, std::string>& brokerAddrs)
+      : cluster_(cluster), broker_name_(brokerName), broker_addrs_(brokerAddrs) {}
+
+  void encode(ProtobufWkt::Struct& data_struct);
+
+  const std::string& cluster() const { return cluster_; }
+
+  const std::string& brokerName() const { return broker_name_; }
+
+  std::unordered_map<int64_t, std::string>& brokerAddresses() { return broker_addrs_; }
+
+private:
+  std::string cluster_;
+  std::string broker_name_;
+  std::unordered_map<int64_t, std::string> broker_addrs_;
+};
+
+class TopicRouteData {
+public:
+  void encode(ProtobufWkt::Struct& data_struct);
+
+  TopicRouteData() = default;
+
+  TopicRouteData(std::vector<QueueData>& queueData, std::vector<BrokerData>& brokerData)
```

Comment:
please replace `std::vector<QueueData>& queueData` to `std::vector<QueueData>&& queueData`

It is not easy to read
```
std::vector<QueueData> test
TopicRouteData tt(test);
// The following code is not aware that test has been moved and will continue to use test
```
But it's much easier to read if you go like this

```
std::vector<QueueData> test
TopicRouteData tt(std::move(test));
// I knew from looking at the code that test had been moved.
```

Reply:
OK, let's use the move way.